src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -0,0 +1,133 @@
+package org.hammerlab.guacamole.commands
+
+import org.apache.spark.SparkContext
+import org.apache.spark.util.StatCounter
+import org.hammerlab.guacamole.{ DistributedUtil, SparkCommand, Common }
+import org.hammerlab.guacamole.reads.{ Read, MappedRead }
+import org.kohsuke.args4j.{ Option => Args4JOption }
+
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Structural Variant caller
+ *
+ * This currently looks for regions with abnormally large insert sizes. This typically indicates
+ * that there's been a large deletion.
+ *
+ * The output is a text file of genomic ranges where there's some evidence of an SV.
+ */
+object StructuralVariant {
+
+  val MAX_INSERT_SIZE = 25000
+  val BLOCK_SIZE = 25
+
+  protected class Arguments extends Common.Arguments.Reads with DistributedUtil.Arguments {
+    @Args4JOption(name = "--filter-contig", usage = "Filter to alignments where either mate is in this contig.")
+    var filterContig: String = ""
+
+    @Args4JOption(name = "--output", usage = "Path to output text file.")
+    var output: String = ""
+  }
+
+  case class Location(
+    contig: String,
+    position: Long) {
+  }
+
+  case class GenomeRange(
+    contig: String,
+    start: Long,
+    stop: Long
+  )
+
+  object Caller extends SparkCommand[Arguments] {
+    override val name = "structural-variant"
+    override val description = "Find structural variants, i.e. large insertions or deletions, inversions and translocations."
+
+    def isFirstInPair(read: MappedRead) = {
+      read.matePropertiesOpt match {
```

Comment:
Is this a reasonable way to get at `isFirstInPair`, or is there a simpler way to do it with `Option` types?